src/ForeignController.sol

```diff
@@ -169,6 +190,41 @@ contract ForeignController is AccessControl {
         emit CentrifugeRecipientSet(destinationCentrifugeId, recipient);
     }
 
+
+    function setMaxSlippage(address pool, uint256 maxSlippage) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        maxSlippages[pool] = maxSlippage;
+        emit MaxSlippageSet(pool, maxSlippage);
+    }
+
+    function setUniswapV3PositionManager(address positionManager) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(positionManager != address(0), "ForeignController/invalid-position-manager");
+
+        uniswapV3PositionManager = INonfungiblePositionManager(positionManager);
+        emit UniswapV3PositionManagerSet(positionManager);
+    }
+
+    function setUniswapV3AddLiquidityLowerTickBound(address pool, int24 lowerTickBound) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        UniswapV3Lib.UniswapV3PoolParams storage params = uniswapV3PoolParams[pool];
+        require(lowerTickBound >= MIN_TICK && lowerTickBound < params.addLiquidityTickBounds.upper, "MainnetController/lower-tick-out-of-bounds");
+
+        params.addLiquidityTickBounds.lower = lowerTickBound;
+        emit UniswapV3PoolParamsUpdated(pool, params);
+    }
+
+    function setUniswapV3AddLiquidityUpperTickBound(address pool, int24 upperTickBound) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        UniswapV3Lib.UniswapV3PoolParams storage params = uniswapV3PoolParams[pool];
+        require(upperTickBound > params.addLiquidityTickBounds.lower && upperTickBound <= MAX_TICK, "MainnetController/upper-tick-out-of-bounds");
+
+        params.addLiquidityTickBounds.upper = upperTickBound;
+        emit UniswapV3PoolParamsUpdated(pool, params);
+    }
```

Comment:
Do we care that `lowerTickBound` is negative and `upperTickBound` stays positive? If so, we should add above and below zero checks to these.

Reply:
midpoint ticks can be in the low negatives (if `token0` is 18-precision and `token1` is 6-precision) or high positives (ie the reverse)